Dockerfile

```diff
@@ -10,6 +10,7 @@ WORKDIR /go/src/github.com/docker/cli
 
 RUN git clone https://github.com/docker/cli . && git checkout a1b83ffd2cbeefc0752e5aa7a543d49c1ddfd2cb
 
+ARG GOPROXY=direct
```

Comment:
Significant issue I see here is that this forces a value to GOPROXY, not just let us override the default one on purpose. Any suggestions for adequate docker build hack is welcome :)

Reply:
Same as for the command-line argument, you can omit the value (and `=`);


## 1. no `--build-arg`

```bash
docker build -<<'EOF'
FROM busybox
ARG GOPROXY
RUN env && exit 1
EOF

 > [2/2] RUN env && exit 1:
#5 0.563 SHLVL=1
#5 0.563 HOME=/root
#5 0.563 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
#5 0.563 PWD=/
```

## 2. with `--build-arg=GOPROXY`, but no env-var set 

```bash
docker build --build-arg=GOPROXY -<<'EOF'
FROM busybox
ARG GOPROXY
RUN env && exit 1
EOF

 > [2/2] RUN env && exit 1:
#4 0.329 SHLVL=1
#4 0.329 HOME=/root
#4 0.329 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
#4 0.329 PWD=/
```

## 3. with `--build-arg=GOPROXY` and `$GOPROXY` set as env-var


```bash
GOPROXY=direct docker build --build-arg=GOPROXY -<<'EOF'
FROM busybox
ARG GOPROXY
RUN env && exit 1
EOF

 > [2/2] RUN env && exit 1:                                                                                                                                                                                                                      
#5 0.376 SHLVL=1                                                                                                                                                                                                                                 
#5 0.376 HOME=/root                                                                                                                                                                                                                              
#5 0.376 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                                                                                                       
#5 0.376 GOPROXY=direct
#5 0.376 PWD=/
```

## 4. with value passed as `--build-arg=GOPROXY=foobar`


```bash
docker build --build-arg=GOPROXY=foobar -<<'EOF'
FROM busybox
ARG GOPROXY
RUN env && exit 1
EOF

 > [2/2] RUN env && exit 1:                                                                                                                                                                                                                      
#5 0.327 SHLVL=1                                                                                                                                                                                                                                 
#5 0.327 HOME=/root                                                                                                                                                                                                                              
#5 0.327 PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin                                                                                                                                                                       
#5 0.327 GOPROXY=foobar
#5 0.327 PWD=/
```